usr/local/www/services_captiveportal_zones_edit.php

```diff
@@ -84,40 +84,36 @@
 		exit;
 	}
 }
+
 include("head.inc");
-?>
-<body link="#0000CC" vlink="#0000CC" alink="#0000CC">
-<?php include("fbegin.inc"); ?>
-<?php if ($input_errors) print_input_errors($input_errors); ?>
-	<form action="services_captiveportal_zones_edit.php" method="post" name="iform" id="iform">
-		<table width="100%" border="0" cellpadding="6" cellspacing="0" summary="captive portal edit">
-			<tr>
-				<td colspan="2" valign="top" class="listtopic"><?=gettext("Edit Captive Portal Zones");?></td>
-			</tr>
-			<tr>
-				<td width="22%" valign="top" class="vncellreq"><?=gettext("Zone name"); ?></td>
-				<td width="78%" class="vtable">
-					<input name="zone" type="text" class="formfld unknown" id="zone" size="64" />
-					<br />
-					<span class="vexpl"><?=gettext("Zone name. Can only contain letters, digits, and underscores (_)."); ?></span>
-				</td>
-			</tr>
-			<tr>
-				<td width="22%" valign="top" class="vncell"><?=gettext("Description"); ?></td>
-				<td width="78%" class="vtable">
-					<input name="descr" type="text" class="formfld unknown" id="descr" size="40" />
-					<br />
-					<span class="vexpl"><?=gettext("You may enter a description here for your reference (not parsed)"); ?>.</span>
-				</td>
-			</tr>
-			<tr>
-				<td width="22%" valign="top">&nbsp;</td>
-				<td width="78%">
-					<input name="Submit" type="submit" class="formbtn" value="<?=gettext("Continue"); ?>" />
-				</td>
-			</tr>
-		</table>
-	</form>
-<?php include("fend.inc"); ?>
-</body>
-</html>
+
+if ($input_errors)
+	print_input_errors($input_errors);
+
+require('classes/Form.class.php');
+
+$form = new Form(new Form_Button(
+	'submit',
+	gettext("Continue")
+));
+
+$section = new Form_Section('Edit Captive Portal Zones');
+
+$section->addInput(new Form_Input(
+	'zone',
+	'Zone name',
+	'text'
+))->setHelp('
+Zone name. Can only contain letters, digits, and underscores (_).');
```

Comment:
Please enforce this guideline by doing $input->setPattern. In this case the argument would be '[0-9a-z_]+'

http://www.w3.org/TR/2011/WD-html5-20110525/common-input-element-attributes.html#the-pattern-attribute